compiler/damlc/tests/src/DA/Test/IncrementalPackageDb.hs

```diff
@@ -0,0 +1,273 @@
+        [ test
+              "Reinitialize when dependency changes"
+              [ ( "daml.yaml"
+                , unlines
+                      [ "sdk-version: 0.0.0"
+                      , "name: proj"
+                      , "source: daml"
+                      , "version: 0.0.1"
+                      , "dependencies:"
+                      , "  - daml-prim"
+                      , "  - daml-stdlib"
+                      , "  - dependency/dep.dar"
+                      ])
+              , ("daml/A.daml"
+                , unlines
+                      [ "module A where"
+                      , "import B"
+                      , "x : Int"
+                      , "x = f 1"])
+              ]
+              []
+              [ ( "daml.yaml"
+                , unlines
+                      [ "sdk-version: 0.0.0"
+                      , "name: dep"
+                      , "source: daml"
+                      , "version: 0.0.1"
+                      , "dependencies:"
+                      , "  - daml-prim"
+                      , "  - daml-stdlib"
+                      ])
+              , ( "daml/B.daml"
+                , unlines
+                      [ "module B where"
+                      , "template T"
+                      , "  with"
+                      , "    p: Party"
+                      , "  where"
+                      , "    signatory p"
+                      , "f : Int -> Int"
+                      , "f i = i + 1"
+                      ])
+              ]
+              [ ( "daml/B.daml"
+                , unlines
+                      [ "module B where"
+                      , "template T"
+                      , "  with"
+                      , "    p: Party"
+                      , "  where"
+                      , "    signatory p"
+                      , "f : Int -> Int"
+                      , "f i = i + 2"
+                      ])
+              ]
+              (ExpectReinitialization True)
+              (ShouldSucceed True)
+        , test
+              "Reinitialize when dependency is added"
+              [ ( "daml.yaml"
+                , unlines
+                      [ "sdk-version: 0.0.0"
+                      , "name: proj"
+                      , "source: daml"
+                      , "version: 0.0.1"
+                      , "dependencies:"
+                      , "  - daml-prim"
+                      , "  - daml-stdlib"
+                      ])
+              , ("daml/A.daml", unlines ["module A where"])
+              ]
+              [ ( "daml.yaml"
+                , unlines
+                      [ "sdk-version: 0.0.0"
+                      , "name: proj"
+                      , "source: daml"
+                      , "version: 0.0.1"
+                      , "dependencies:"
+                      , "  - daml-prim"
+                      , "  - daml-stdlib"
+                      , "  - dependency/dep.dar"
+                      ])
```

Comment:
maybe reference something from `B` to make sure everything really worked correctly and we didn’t just write the dep again?

Reply:
added a reference.